src/store/messages/saga.ts

```diff
@@ -277,16 +277,19 @@ export function* stopSyncChannels(action) {
 
 export function* receiveNewMessage(action) {
   const { channelId, message } = action.payload;
+  const channelIdWithoutPrefix = channelId.replace('sendbird_group_channel_', '');
 
-  const cachedMessageIds = [...(yield select(getCachedMessageIds(channelId)))];
-  const currentMessages = yield select(rawMessagesSelector(channelId));
+  const cachedMessageIds = [...(yield select(getCachedMessageIds(channelIdWithoutPrefix)))];
+  const currentMessages = yield select(rawMessagesSelector(channelIdWithoutPrefix));
 
   let messages = [];
+  let cachedMessageLength = 0;
```

Comment:
I might be missing something but why do we need this variable? It seems to me we changed an if/else statement into an "if...and then also do"...why not just move the "also" part into the first if statement? (and was the `else` part of the statement the bug?)